src/lib/swapper/api.ts

```diff
@@ -274,3 +280,33 @@ export interface Swapper<T extends ChainId, MaybeUnknownNetworkFee extends boole
    */
   getTradeTxs(tradeResult: TradeResult): Promise<Result<TradeTxs, SwapErrorRight>>
 }
+
+export type UnsignedTx = SignTx<keyof ChainSignTx>
+
+export type TradeQuote2 = TradeQuote & { id: string; receiveAddress: string; affiliateBps: string }
+
+export type GetUnsignedTxArgs = {
+  tradeQuote: TradeQuote2
+  chainId?: ChainId
+  accountMetadata?: AccountMetadata
+  stepIndex: number
+  supportsEIP1559: boolean
+} & ({ from: string; xpub?: never } | { from?: never; xpub: string })
+
+export type ExecuteTradeArgs = {
+  txToExecute: UnsignedTx
+  wallet: HDWallet
+  chainId: ChainId
+}
+
+export type Swapper2 = {
+  getTradeQuote: (
+    input: GetTradeQuoteInput,
+    ...deps: any[]
```

Comment:
Can we type these?

Reply:
Or I suppose this is on purpose so that consumers can type their additional arguments at position 0+